src/CommonITILObject.php

```diff
@@ -7130,7 +7121,7 @@ class_exists($validation_class) && $params['with_validations']
 
                 $item['timeline_position'] = $document_item['timeline_position'];
                 $item['_can_edit'] = $can_view_documents && $document_obj->canUpdateItem();
-                $item['_can_delete'] = $can_view_documents && $document_obj->canDeleteItem();
+                $item['_can_delete'] = $can_view_documents && $document_obj->canDeleteItem() && $canupdate_parent;
```

Comment:
This ensure that `delete` button on a document is not available when ticket is closed.